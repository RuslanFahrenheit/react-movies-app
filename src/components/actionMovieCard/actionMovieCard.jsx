import React from 'react';
import PropTypes from 'prop-types';
import { Select } from '../select';
import { Input } from '../input';
import { Button } from '../button';

const ActionMovieCard = ({
  form,
  submitBtnText = 'submit',
  cancelBtnText = 'cancel',
  handleSubmit,
  handleCancel,
}) => (
  <>
    {form.map(({
      id,
      label,
      placeholder,
      type,
      options,
    }) => (
      (type === 'select')
        ? (
          <Select
            key={id}
            options={options}
            selectId={id}
            label={label}
          />
        )
        : (
          <Input
            key={id}
            label={label}
            placeholder={placeholder}
            type={type}
          />
        )
    ))}
    <Button
      variant="secondary"
      onClick={handleSubmit}
    >
      {cancelBtnText}
    </Button>
    <Button
      variant="primary"
      onClick={handleCancel}
    >
      {submitBtnText}
    </Button>
  </>
);

const {
  instanceOf,
  string,
  func,
} = PropTypes;

ActionMovieCard.propTypes = {
  form: instanceOf(Array).isRequired,
  handleSubmit: func,
  handleCancel: func,
  submitBtnText: string,
  cancelBtnText: string,
};

export { ActionMovieCard };
