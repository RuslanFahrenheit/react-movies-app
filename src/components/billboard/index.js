export { default as Billboard } from './billboard';
