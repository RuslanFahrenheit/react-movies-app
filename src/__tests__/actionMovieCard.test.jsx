import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { ActionMovieCard } from '../components/actionMovieCard';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render a heading', () => {
  act(() => {
    render(
      <ActionMovieCard
        form={{}}
        submitBtnText="submit"
        cancelBtnText="reset"
        handleSubmit={() => {}}
        handleCancel={() => {}}
      />,
      container,
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
