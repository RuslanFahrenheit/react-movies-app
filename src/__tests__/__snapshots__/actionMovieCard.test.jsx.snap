// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a heading 1`] = `
"<form><label class=\\"label\\">title<input class=\\"input input--text\\" type=\\"text\\" name=\\"title\\" placeholder=\\"Select title\\" value=\\"\\"></label><label class=\\"label\\">release date<input class=\\"input input--text\\" type=\\"text\\" name=\\"year\\" placeholder=\\"Select date\\" value=\\"\\"></label><label class=\\"label\\">movie url<input class=\\"input input--text\\" type=\\"text\\" name=\\"url\\" placeholder=\\"Movie URL here\\" value=\\"\\"></label><label for=\\"\\">genre</label><select class=\\"select\\" id=\\"\\">
    <option class=\\"select__option\\" value=\\"drama\\">Drama</option>
    <option class=\\"select__option\\" value=\\"comedy\\">Comedy</option>
  </select><label class=\\"label\\">overview<input class=\\"input input--text\\" type=\\"text\\" name=\\"overview\\" placeholder=\\"Overview here\\" value=\\"\\"></label><label class=\\"label\\">runtime<input class=\\"input input--text\\" type=\\"text\\" name=\\"runtime\\" placeholder=\\"Runtime here\\" value=\\"\\"></label><button class=\\"button button--secondary\\" type=\\"button\\">reset</button><button class=\\"button button--primary\\" type=\\"submit\\">submit</button></form>"
`;
